pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,159 @@
+* `images_folder` - folder containing input images
+* `split_file` - the most important part of the dataset. This file contains `file_name` (tab symbol) `formula_idx` per line connecting corresponding index of the formula in the file with formulas and particular image with `image_name`. Example:
+    ```
+    11.png  11
+    34.png  34
+    ```
+    There should be at least two such files: `train_filter.lst` and `validate_filter.lst`
+
+> **NOTE**:
+> By default code asserts to see dataset structure with the following names:
+> `images_processed` - folder with images
+> `formulas.norm.lst` - file with preprocessed formulas, for details, refer to [im2markup](https://github.com/harvardnlp/im2markup) repository
+> `validate_filter.lst` and `train_filter.lst` - corresponding splits of the data.
+
+
+## Training
+
+To train Text Spotter model run:
+
+```bash
+python3 tools/train.py --config configs/train_config.yml --work_dir <path to work dir>
+```
+Work dir is used to store information about learning: saved model checkpoints, logs.
+
+### Description of the possible option in train config:
+ - `backbone_config`:
+    * `arch`: type of the architecture (if backbone_type is resnet). For mor details, please, refer to [ResnetLikeBackBone](im2latex/models/backbones/resnet.py)
+    * `disable_layer_3` and
+    * `disable_layer_4` - disables layer 3 and 4 in resnetlike backbone
+    * `enable_last_conv` - enables additional convolution layer to adjust number of output channels to the number of input channels in the LSTM
+    * `in_lstm_ch` - number of input LSTM channels, used for `last_conv`
+- `backbone_type`: `resnet` for resnetlike backbone or anything else for original backbone from [im2markup](https://arxiv.org/pdf/1609.04938.pdf) paper
+- `batch_size` - batch size used for training
+- `device` - device for training, used in pytorch .to() method. Possible options: 'cuda', 'cpu', etc. `cpu` is used by default.
+- `head` - configuration of the text recognition head
+    * `beam_width` - witdth used in beam search. 0 - do not use beam search, 1 and more - use beam search with corresponding number of possible tracks.
+    * `dec_rnn_h` - number of channels on decoding
+    * `emb_size` - dimension of the embedding
+    * `enc_rnn_h` - number of channels in encoding
+    * `in_lstm_ch` - number of input in lstm channels, should be equal to `backbone_config.in_lstm_ch`
+    * `max_len` - maximum possible length of the formula
+    * `n_layer` - describe
+- `learning_rate` - learining rate
+- `log_path` - path to store training logs
+- `model_path` - path for model (if one wants to aftertune model)
+- `optimizer` - Adam or SGD
+- `save_dir` - dir to save checkpoints
+- `train_paths` - list of paths from where get training data (if more than one path is specified, datasets are concatenated). If one wants to concatenate more than one instance of the desirable dataset, this dataset shoulb be set several times.
```

Comment:
`this dataset shoulb be set several times.` what?

Reply:
fixed